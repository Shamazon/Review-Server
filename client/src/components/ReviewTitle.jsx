
import React from 'react';
import moment from 'moment';
import style from '../../dist/style/style.css'

var ReviewTitle = (props) => {

    var stars = [];
    for(var i = 0; i < 5; i++){
        if(props.review.star_rating >= i) {
            stars.push(<img key={i} className={style.star} src='./img/fullstar.png' />)
        } else {
            stars.push(<img key={i} className={style.star} src='./img/emptystar.png' />)
        }
    }

    return (
        <div>
            <span>{stars}</span>
            <a className={`${style.reviewTitle} ${style.underline}`}>{props.review.title}</a>   
            
            <div className={style.gray}>
                {moment().subtract(1, 'years') < moment(props.review.date) ?
                 moment(props.review.date).fromNow(): 
                 moment(props.review.date).format('MMMM Do YYYY') 
                }</div>
        </div>
    )
}

export default ReviewTitle;