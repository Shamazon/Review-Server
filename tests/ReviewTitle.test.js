import React from 'react';
import ReviewTitle from '../client/src/components/ReviewTitle.jsx';
import enzyme from 'enzyme';
import moment from 'moment';

describe('ReviewTitle', () => {
  it('should be defined', () => {
    expect(ReviewTitle).toBeDefined();
  });
  it('should render correctly', () => {
    const props = {
      review: {
        starRating: 5,
        title: 'test',
        date: moment('2017-09-15 09:30:00').format('MMM Do YYYY h:mm:ss a'),
      },
    };
    const tree = enzyme.shallow(<ReviewTitle {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
