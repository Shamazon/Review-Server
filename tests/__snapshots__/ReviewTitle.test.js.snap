// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewTitle should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewTitle
    review={
      Object {
        "date": "Sep 15th 2017 9:30:00 am",
        "starRating": 5,
        "title": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          <img
            className="star"
            src="./img/emptystar.png"
          />
          <img
            className="star"
            src="./img/emptystar.png"
          />
          <img
            className="star"
            src="./img/emptystar.png"
          />
          <img
            className="star"
            src="./img/emptystar.png"
          />
          <img
            className="star"
            src="./img/emptystar.png"
          />
        </span>,
        <a
          className="reviewTitle underline"
        >
          test
        </a>,
        <div
          className="gray"
        >
          Invalid date
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="star"
              src="./img/emptystar.png"
            />,
            <img
              className="star"
              src="./img/emptystar.png"
            />,
            <img
              className="star"
              src="./img/emptystar.png"
            />,
            <img
              className="star"
              src="./img/emptystar.png"
            />,
            <img
              className="star"
              src="./img/emptystar.png"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "className": "star",
              "src": "./img/emptystar.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "className": "star",
              "src": "./img/emptystar.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "className": "star",
              "src": "./img/emptystar.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "className": "star",
              "src": "./img/emptystar.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "className": "star",
              "src": "./img/emptystar.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "reviewTitle underline",
        },
        "ref": null,
        "rendered": "test",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Invalid date",
          "className": "gray",
        },
        "ref": null,
        "rendered": "Invalid date",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <img
              className="star"
              src="./img/emptystar.png"
            />
            <img
              className="star"
              src="./img/emptystar.png"
            />
            <img
              className="star"
              src="./img/emptystar.png"
            />
            <img
              className="star"
              src="./img/emptystar.png"
            />
            <img
              className="star"
              src="./img/emptystar.png"
            />
          </span>,
          <a
            className="reviewTitle underline"
          >
            test
          </a>,
          <div
            className="gray"
          >
            Invalid date
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="star"
                src="./img/emptystar.png"
              />,
              <img
                className="star"
                src="./img/emptystar.png"
              />,
              <img
                className="star"
                src="./img/emptystar.png"
              />,
              <img
                className="star"
                src="./img/emptystar.png"
              />,
              <img
                className="star"
                src="./img/emptystar.png"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "className": "star",
                "src": "./img/emptystar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "className": "star",
                "src": "./img/emptystar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "className": "star",
                "src": "./img/emptystar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "className": "star",
                "src": "./img/emptystar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "className": "star",
                "src": "./img/emptystar.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "reviewTitle underline",
          },
          "ref": null,
          "rendered": "test",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Invalid date",
            "className": "gray",
          },
          "ref": null,
          "rendered": "Invalid date",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
